mlir/include/mlir/Dialect/EmitC/IR/EmitC.td

```diff
@@ -288,6 +288,7 @@ def EmitC_CastOp : EmitC_Op<"cast",
   let arguments = (ins EmitCType:$source);
   let results = (outs EmitCType:$dest);
   let assemblyFormat = "$source attr-dict `:` type($source) `to` type($dest)";
+  let hasFolder = 1;
```

Comment:
Why is this needed? I would have thought folding is done through the CastOpInterface, but I haven't checked.

Reply:
Good catch -- looks like this folder works. I don't know why it didn't happen during my tests...